sdk/src/main/java/com/horizen/account/secret/PrivateKeySecp256k1.java

```diff
@@ -43,7 +44,8 @@ public SecretSerializer serializer() {
     @Override
     public AddressProposition publicImage() {
         var publicKey = ECKeyPair.create(privateKey).getPublicKey();
-        return new AddressProposition(Numeric.toBytesPadded(publicKey, Secp256k1.PUBLIC_KEY_SIZE));
+        var hashedKey = Hash.sha3(Numeric.toBytesPadded(publicKey, Secp256k1.PUBLIC_KEY_SIZE));
+        return new AddressProposition(Arrays.copyOfRange(hashedKey, 0, 20));
```

Comment:
I just doubled checked: It must be the **last** 20 bytes, not the first:
https://ethereum.org/en/developers/docs/accounts/#:~:text=The%20public%20key%20is%20generated,adding%200x%20to%20the%20beginning.

GETH implementation:
```go
func PubkeyToAddress(p ecdsa.PublicKey) common.Address {
	pubBytes := FromECDSAPub(&p)
	return common.BytesToAddress(Keccak256(pubBytes[1:])[12:])
}
```
(Ignore the one-byte-skip `pubBytes[1:]` that is just because they use the standard format for ec keys where the first byte contains a marker for the type of key - in this case always 0x04 for uncompressed keys)

Also using the `Account.ADDRESS_SIZE` constant would be nice :wink: